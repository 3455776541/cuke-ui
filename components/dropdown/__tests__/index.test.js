import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Dropdown from "../index";

const overlay = (
  <ul>
    <li> 红烧牛肉 </li>
    <li> 麻辣干锅 </li>
    <li> 仔姜肉丝 </li>
  </ul>
);

describe("<Dropdown/>", () => {
  it("should render a <Dropdown/> components", () => {
    const wrapper = render(
      <div>
        <Dropdown overlay={overlay}>
          <a href="#">今日菜单(有动画)</a>
        </Dropdown>
        <Dropdown overlay={overlay} animate={false}>
          <a href="#">今日菜单(无动画)</a>
        </Dropdown>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-dropdown classnames", () => {
    const wrapper = render(
      <div>
        <Dropdown overlay={overlay}>
          <a href="#">今日菜单(有动画)</a>
        </Dropdown>
        <Dropdown overlay={overlay} animate={false}>
          <a href="#">今日菜单(无动画)</a>
        </Dropdown>
      </div>
    );
    assert(wrapper.find(".cuke-dropdown").length >= 1);
    assert(wrapper.find(".cuke-dropdown-wrap").length >= 1);
    assert(wrapper.find(".cuke-dropdown-overlay").length >= 1);
  });

  it("should render disabled", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <div>
        <Dropdown overlay={overlay} disabled onVisibleChange={onChange}>
          <a href="#">今日菜单(有动画)</a>
        </Dropdown>
      </div>
    );
    wrapper.find("div").simulate("change");
    expect(onChange).not.toHaveBeenCalled();
  });

  it("should render many animate type", () => {
    const wrapper = render(
      <div>
        <Dropdown overlay={overlay} animate="slideUp">
          slideDown
        </Dropdown>
        <Dropdown overlay={overlay} animate="slideDown">
          slideDown
        </Dropdown>
        <Dropdown overlay={overlay} animate="slideRight">
          slideRight
        </Dropdown>

        <Dropdown overlay={overlay} animate="slideLeft">
          slideLeft
        </Dropdown>

        <Dropdown overlay={overlay} animate={false}>
          no animate
        </Dropdown>
      </div>
    );
    assert(wrapper.find(".cuke-dropdown-overlay-slideUp").length === 1);
    assert(wrapper.find(".cuke-dropdown-overlay-slideDown").length === 1);
    assert(wrapper.find(".cuke-dropdown-overlay-slideRight").length === 1);
    assert(wrapper.find(".cuke-dropdown-overlay-slideLeft").length === 1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should trigger on visible change when hover", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Dropdown overlay={overlay} onVisibleChange={onChange}>
        <a href="#">今日菜单(有动画)</a>
      </Dropdown>
    );
    wrapper.find(".cuke-dropdown").simulate("blur");
    expect(onChange).toHaveBeenCalled();
  });

  it("should trigger on visible change when clicked", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Dropdown overlay={overlay} onVisibleChange={onChange} trigger="click">
        <a href="#">今日菜单(有动画)</a>
      </Dropdown>
    );
    wrapper.find(".cuke-dropdown").simulate("focus");
    expect(onChange).toHaveBeenCalled();
  });
  it("should trigger on visible change when clicked", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Dropdown overlay={overlay} onVisibleChange={onChange} trigger="click">
        <a href="#">今日菜单(有动画)</a>
      </Dropdown>
    );
    wrapper.find(".cuke-dropdown").simulate("blur");
    expect(onChange).toHaveBeenCalled();
  });
});
