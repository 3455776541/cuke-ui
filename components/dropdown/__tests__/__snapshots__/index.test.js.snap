// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown/> should render a <Dropdown/> components 1`] = `
<div>
  <div
    class="cuke-dropdown"
  >
    <div
      class="cuke-dropdown-wrap"
    >
      <a
        href="#"
      >
        今日菜单(有动画)
      </a>
    </div>
    <div
      class="cuke-dropdown-overlay cuke-dropdown-overlay-slideUp"
    >
      <ul>
        <li>
           红烧牛肉 
        </li>
        <li>
           麻辣干锅 
        </li>
        <li>
           仔姜肉丝 
        </li>
      </ul>
    </div>
  </div>
  <div
    class="cuke-dropdown"
  >
    <div
      class="cuke-dropdown-wrap"
    >
      <a
        href="#"
      >
        今日菜单(无动画)
      </a>
    </div>
    <div
      class="cuke-dropdown-overlay cuke-dropdown-overlay-false"
    >
      <ul>
        <li>
           红烧牛肉 
        </li>
        <li>
           麻辣干锅 
        </li>
        <li>
           仔姜肉丝 
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<Dropdown/> should render many animate type 1`] = `
<div>
  <div
    class="cuke-dropdown"
  >
    <div
      class="cuke-dropdown-wrap"
    >
      slideDown
    </div>
    <div
      class="cuke-dropdown-overlay cuke-dropdown-overlay-slideUp"
    >
      <ul>
        <li>
           红烧牛肉 
        </li>
        <li>
           麻辣干锅 
        </li>
        <li>
           仔姜肉丝 
        </li>
      </ul>
    </div>
  </div>
  <div
    class="cuke-dropdown"
  >
    <div
      class="cuke-dropdown-wrap"
    >
      slideDown
    </div>
    <div
      class="cuke-dropdown-overlay cuke-dropdown-overlay-slideDown"
    >
      <ul>
        <li>
           红烧牛肉 
        </li>
        <li>
           麻辣干锅 
        </li>
        <li>
           仔姜肉丝 
        </li>
      </ul>
    </div>
  </div>
  <div
    class="cuke-dropdown"
  >
    <div
      class="cuke-dropdown-wrap"
    >
      slideRight
    </div>
    <div
      class="cuke-dropdown-overlay cuke-dropdown-overlay-slideRight"
    >
      <ul>
        <li>
           红烧牛肉 
        </li>
        <li>
           麻辣干锅 
        </li>
        <li>
           仔姜肉丝 
        </li>
      </ul>
    </div>
  </div>
  <div
    class="cuke-dropdown"
  >
    <div
      class="cuke-dropdown-wrap"
    >
      slideLeft
    </div>
    <div
      class="cuke-dropdown-overlay cuke-dropdown-overlay-slideLeft"
    >
      <ul>
        <li>
           红烧牛肉 
        </li>
        <li>
           麻辣干锅 
        </li>
        <li>
           仔姜肉丝 
        </li>
      </ul>
    </div>
  </div>
  <div
    class="cuke-dropdown"
  >
    <div
      class="cuke-dropdown-wrap"
    >
      no animate
    </div>
    <div
      class="cuke-dropdown-overlay cuke-dropdown-overlay-false"
    >
      <ul>
        <li>
           红烧牛肉 
        </li>
        <li>
           麻辣干锅 
        </li>
        <li>
           仔姜肉丝 
        </li>
      </ul>
    </div>
  </div>
</div>
`;
